pallets/stateful-storage/src/types.rs

```diff
@@ -449,30 +585,32 @@ impl<T: Config> ItemizedOperations<T> for ItemizedPage<T> {
 	fn try_parse(&self, include_header: bool) -> Result<ParsedItemPage, PageError> {
 		let mut count = 0u16;
 		let mut items = BTreeMap::new();
-		let mut offset = 0;
-		while offset < self.data.len() {
-			ensure!(
-				offset + ItemHeader::max_encoded_len() <= self.data.len(),
-				PageError::ErrorParsing("wrong header size")
-			);
-			let header = <ItemHeader>::decode(&mut &self.data[offset..])
-				.map_err(|_| PageError::ErrorParsing("decoding header"))?;
-			let item_total_length = ItemHeader::max_encoded_len() + header.payload_len as usize;
+		let mut page_slice = &self.data[..];
+
+		while page_slice.len() > 0 {
+			let item_slice = page_slice;
+			let header = <ItemHeader>::decode(&mut page_slice)
```

Comment:
So the idea is that the decode modifies the page_slice and consumes the bytes that is decoded and it leaves the rest?